File: 220/js/strategy/encoder_spec.js
require('./encoder')

describe("Reverse Encoding", function() {
  it("reverses each word in the input", function() {
	var encoder = new Encoder(new ReverseEncoding());
    expect(encoder.encode("abcde 123")).toEqual("edcba 321");
  });
});

describe("Double Encoding", function() {
  it("repeats each word in the input", function() {
	var encoder = new Encoder(new DoubleEncoding());
    expect(encoder.encode("abcde 123")).toEqual("abcdeabcde 123123");
  });
});

File: 220/ruby/strategy/encoder_spec.rb
require './encoder'

describe ReverseEncoding do
  it "reverses each word in the input" do
    encoder = Encoder.new(ReverseEncoding.new)
    encoder.encode("abcde 123").should == "edcba 321" 
  end
end

describe DoublingEncoding do
  it "repeats each word in the input" do
    encoder = Encoder.new(DoublingEncoding.new)
    encoder.encode("abcde 123").should == "abcdeabcde 123123" 
  end
end